### ISSUE TITLE
[Suggestion] Handheld terminals!

### ISSUE THREAD

**Issue by ozin370** [KSP-KOS/KOS#1767]
Now that we have intervessel communication I think it would be great to give Kerbals in EVA their own terminals. Possibly integrated with KIS?

Thoughts?

**Comment by GER-Space** [comment]
I like the Idea. It could be possible to add the kOS-processor partmodule to a KIS part with a ModuleManager patch. 
My hopes are low, that it might be that easy, because kOS asumes in some functions, that it is installed on a vessel and I have no Idea if a Kerbal is enough vessel-like for kOS to boot up.

Things that will never work: 

- Cooked Steering. 
- Throttle Control
- SAS

Things that might work: 

- raw steering 
- raw steering with rcs
- interaction with other vessel parts
- take a surface sample / EVA report.

Things need new Code (via addon or directly within kOS): 

- start a EVA
- grab / release
- set movement speed
- planting a flag.
- KIS interactions
- all KAS actions
- jumping

**Comment by ozin370** [comment]
Yeah, obviously some features would need to be disabled. I was mostly thinking of the benefits of being able to use the new communication features with an EVA'd kerbal (with a fairly limited range).

Actually steering the kerbal would be a bit strange to me, wouldn't exactly fit with what kOS has been doing so far. Same goes with KIS/KAS interaction. That is of course unless the devs decide to go down the android route.. lol

**Comment by GER-Space** [comment]
OK, here we go: 

- a small kOS computer with unlimited power. (no EC consumers)
- 2km RemoteTech omni antenna . 
- Science is working (eva report / surface sample)
- Kerbal doesn't want to move. 
- interactions with other parts are possible. (Actions and Events)

[kOS-Pad.zip](https://github.com/KSP-KOS/KOS/files/398074/kOS-Pad.zip)

The temporary assets are from wildbluetools.

**Comment by GER-Space** [comment]
After watching a lot of people playing "Pokemon Go" : Controlling Kerbals with a smartphone/tablet doesn't feel so strange anymore. :-)

**Comment by GER-Space** [comment]
I got the permission from Angel-125 for using his kPad model. 

For building Kerbal-Robots: Kerbals cannot be modified through a ModuleManager patch. This requires some coding, but it will turn (all kerbals)/ (one gender) into Stepford Kerbals.

**Comment by Dunbaratu** [comment]
So let's see.  A Kerbal is a vessel composed of one "part", so it should hypothetically be possible to add kOS as a PartModule to the Kerbal "part", but we'd need some config options to put in the "part.cfg"-like settings to disable things that kOS modules normally get to do by default (like lock steering, etc.)

**Comment by GER-Space** [comment]
When I tested the kOS-Pad. (this makes the Kerbal a vessel with two parts) lock steering simply does nothing, but it didn't produce any error. Performing and transmitting science experiments worked.

**Comment by ozin370** [comment]
Oh wow, didn't check this thread until now, didn't dare hope for an implementation of this (at least not this soon), I'll try this out immediately.

**Comment by ozin370** [comment]
Hmm, it seems like user input is really limited, the action groups don't work.

**Comment by GER-Space** [comment]
@ozin370   Perhaps you can use the AGX Addon? 

I made a plugin, so the pad is automatic equipted, when the kerbal is selected (by going eva or by [] ).

In the next few days I hope I can give them a bit more things to to.

**Comment by ozin370** [comment]
The only user inputs while running a script that I have seen working are the user interface buttons found on the top of the window (lights,gear,brakes,abort), no keyboard inputs work sadly. Eventually I would really like to see a possibly customizable kOS panel with buttons and toggles that can be clicked on with the mouse.

**Comment by GER-Space** [comment]
@Dunbaratu I'm already able to set the default bootfilename on initialization of the eva-addon, so it should be possible to change the other parameters. (diskspace, ECperInstruction etc..)

After I made the Kerbal move I'll go on and try to create a kOSBot, which is integrated to the techtree. 
Currently working:

- go on EVA
- Board a part/nearest part
- Calling a event by it's name (like: jumping, grab ladder, release ladder. toggle rcs, Plant a Flag, etc.)
- listing of all event names.
- playing custom animations /created by Kerbal Animation Suite/  (not tested).

WIP:
- making a kerbal turn & move
- playing builtin animations
